maven/detect.go

```diff
@@ -33,8 +36,19 @@ const (
 type Detect struct{}
 
 func (Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
-	file := filepath.Join(context.Application.Path, "pom.xml")
-	_, err := os.Stat(file)
+	l := bard.NewLogger(ioutil.Discard)
+	cr, err := libpak.NewConfigurationResolver(context.Buildpack, &l)
+	if err != nil {
+		return libcnb.DetectResult{}, err
+	}
+
+	pomFile, _ := cr.Resolve("BP_MAVEN_POM_FILE")
+	if pomFile == "" {
+		pomFile = "pom.xml"
+	}
+
+	file := filepath.Join(context.Application.Path, pomFile)
```

Comment:
Can you add a test case to detct_test.go that sets this env variable `BP_MAVEN_POM_FILE` & validates detect passes?

Reply:
:+1: